provider/upstream_provider.go

```diff
@@ -30,6 +33,9 @@ import (
 // runtime startup is somewhat performance sensitive. Therefore any modifications undertaken here should complete
 // quickly.
 func newUpstreamProvider(ctx context.Context) awsShim.UpstreamProvider {
+	log.SetOutput(io.Discard)      // ignore logging from upstream constructors
```

Comment:
I have moved here @iwahbe @flostadler that still works.

This is not entirely the right thing to do. We need to revisit bridge Main APIs and clean them up. The bridge should be initializing the provider and setting up logging against the Pulumi CLI host before calling into the provider-specific constructors. Then instead of discarding these logs would get the regular log sink.

Reply:
We can only get so far with that strategy, but we should do it just to clean up the proliferation of Main APIs.

pulumi-databricks hit log statements during variable setup (before main).